client/state/data-layer/wpcom-http/pipeline/index.js

```diff
@@ -0,0 +1,88 @@
+/**
+ * External dependencies
+ */
+import {
+	compact,
+	pick,
+} from 'lodash';
+
+/**
+ * Internal dependencies
+ */
+import { applyDuplicatesHandlers, removeDuplicateGets } from './remove-duplicate-gets';
+
+/**
+ * @typedef {Object} EgressData
+ * @property {Object} originalRequest the Redux action describing the egressing request
+ * @property {ReduxStore} store Redux store
+ * @property {*} originalData response data from returned network request
+ * @property {*} originalError response error from returned network request
+ * @property {*} nextData transformed response data
+ * @property {*} nextError transformed response error
+ * @property {Object[]} failures list of `onFailure` actions to dispatch
+ * @property {Object[]} successes list of `onSuccess` actions to dispatch
+ * @property {Boolean} [shouldAbort] whether or not no further processing should occur for request
+ */
+
+/**
+ * @typedef {Function} EgressProcessor
+ * @param {EgressData} information about request response at this stage in the pipeline
+ * @returns {EgressData} contains transformed responder actions
+ */
+
+/**
+ * @typedef {Object} IngressData
+ * @property {Object} originalRequest the Redux action describing the ingressing request
+ * @property {ReduxStore} store Redux store
+ * @property {?Object} nextRequest the processed request to pass along the chain
```

Comment:
@samouri I already found a use - if we want to bypass the dedupe phase (perhaps with something like `maxAge` or `isIdempotent` then we want to make sure we're looking at the original request and not a processed form